src/meta/api/src/schema_api_impl.rs

```diff
@@ -1075,6 +1080,35 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
         })
     }
 
+    #[tracing::instrument(level = "debug", ret, err, skip_all)]
+    async fn update_index(&self, req: UpdateIndexReq) -> Result<UpdateIndexReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let index_id_key = IndexId {
+            index_id: req.index_id,
+        };
+
+        let (index_meta_seq, _) = get_index_meta_by_id_or_err(self, &index_id_key).await?;
+
+        if index_meta_seq == 0 {
+            return Err(KVAppError::AppError(AppError::UnknownIndex(
+                UnknownIndex::new(&req.index_name, "update_index"),
+            )));
+        }
+
+        let reply = self
+            .upsert_kv(UpsertKVReq::new(
+                &req.index_id.to_string(),
+                MatchSeq::Exact(index_meta_seq),
+                Operation::Update(serialize_struct(&req.index_meta)?),
+                None,
+            ))
+            .await?;
```

Comment:
It is not a conditional update, i.e., the final state does not depends on the state before update.
Therefore a non-conditional update `upsert_kv(UpsertKVReq::new(.., MatchSeq::GE(1))` would be good enough